src/pcm.c

```diff
@@ -501,7 +501,11 @@ int pcm_set_config(struct pcm *pcm, const struct pcm_config *config)
     memset(&sparams, 0, sizeof(sparams));
     sparams.tstamp_mode = SNDRV_PCM_TSTAMP_ENABLE;
     sparams.period_step = 1;
-    sparams.avail_min = config->period_size;
+    if (!pcm->config.avail_min && (pcm->flags & PCM_MMAP)) {
+        pcm->config.avail_min = sparams.avail_min = pcm->config.period_size;
+    } else {
+        sparams.avail_min = config->avail_min;
+    }
```

Comment:
The original logic is no matter what `pcm->config.avail_min` is, we set `sparams.avail_min` as `config->period_size`. Now the logic becomes when `pcm->config.avail_min` is zero and the stream is a MMAP stream, we set `pcm->config.avail_min` and `sparams.avail_min` as `pcm->config.period_size`. Otherwise, set `sparams.avail_min` as `config->avail_min` even if `pcm->config.avail_min` is zero. 

I'm wondering why can't we make that like

```
    sparams.avail_min = config->period_size;
    if (!pcm->config.avail_min && (pcm->flags & PCM_MMAP)) {
        pcm->config.avail_min = pcm->config.period_size;
    }
```

Reply:
It looks good, let me update.